base/math.jl

```diff
@@ -59,6 +61,7 @@ clamp(x::X, lo::L, hi::H) where {X,L,H} =
                   convert(promote_type(X,L,H), lo),
                   convert(promote_type(X,L,H), x)))
 
+clamp(::Type{T}, x) where {T} = clamp(x, typemin(T), typemax(T)) % T
```

Comment:
Yes.  It doesn't make sense for floats - the `%` operator isn't defined, and the problem doesn't exist because conversion to a narrower float clamps to `typemin` and `typemax` (+/-Inf) anyway.